pandas/plotting/_core.py

```diff
@@ -373,7 +373,7 @@ def boxplot(
         >>> type(boxplot)
         <class 'numpy.ndarray'>
     """
-    plot_backend = _get_plot_backend("matplotlib")
+    plot_backend = _get_plot_backend(kwds.pop("backend", None))
```

Comment:
cc @datapythonista on this one. Do you recall why "matplotlib" was hardcoded here? Wasn't part of the backend API?

Reply:
I"m on my phone and I can't check now, but I think this is the boxplot that is independent from the accessor, `pandas.plotting.boxplot`. There is a proposal to deprecate it, but regardless of that the decision was to only switch the backend for the accessor plots.

Reply:
It was really unclear to me why the matplotlib was hardcoded there. The function seems very similar to `boxplot_frame` and `boxplot_frame_groupby`.

Reply:
#28177 can provide some more context

Reply:
I understand that the API is being unified and all of these boxplot* are being deprecated, which seems like a great plan. I just don't get why the backend should be hardcoded when calling `pandas.plotting.boxplot` and not when calling `df.boxplot` it seems oddly inconsistent.

Reply:
Tom pointed me to https://github.com/pandas-dev/pandas/pull/28159, so I have plenty of explanation now. Feel free to ignore my last comment :)